src/modules/todos.js

```diff
@@ -48,22 +34,111 @@ const renderTodoList = () => {
     const taskDescription = document.createElement('span');
     taskDescription.classList.add('task-description');
     taskDescription.textContent = todo.description;
+
     if (todo.completed) {
       taskDescription.classList.add('completed');
     }
+
     const icon = document.createElement('span');
     icon.classList.add('fas', 'fa-ellipsis-v');
     icon.style.color = '#c8ccd0';
 
+    // Add an event lisnter to the icon
+    icon.addEventListener('click', (e) => {
+      showOptionsMenu(e);
+    });
+
+    function showOptionsMenu(e) {
+      const optionsMenus = document.querySelectorAll('.options-menu');
+      // Hide all other options menus
+      optionsMenus.forEach((menu) => {
+        if (menu !== e.target.querySelector('.options-menu')) {
+          menu.remove();
+        }
+      });
+
+      const optionsMenu = document.createElement('div');
+      optionsMenu.classList.add('options-menu');
+
+      const editButton = document.createElement('button');
+      editButton.textContent = 'Edit';
+      editButton.addEventListener('click', () => {
+        editTodo(todo);
+        hideOptionsMenu();
+      });
+
+      const deleteButton = document.createElement('button');
+      deleteButton.textContent = 'Delete';
+      deleteButton.addEventListener('click', () => {
+        deleteTodo(todo);
+        hideOptionsMenu();
+      });
+
+      optionsMenu.appendChild(editButton);
+      optionsMenu.appendChild(deleteButton);
+      e.target.appendChild(optionsMenu);
+    }
+
+    function hideOptionsMenu() {
+      const optionsMenu = document.querySelector('.options-menu');
+      if (optionsMenu !== null) {
+        optionsMenu.remove();
+      }
+    }
+
+    function editTodo(todo) {
+      const todoIndex = todo.index;
+      input.value = todo.description;
+      addItemButton.innerText = 'Save';
+
+      const saveItem = () => {
+        todos[todoIndex].description = input.value;
+        localStorage.setItem('todos', JSON.stringify(todos));
+        renderTodoList();
+        hideOptionsMenu();
+        addItemButton.innerText = '+';
+        addItemButton.classList.add('plus-on-edit');
+        addItemButton.removeEventListener('click', saveItem);
+      };
+
+      addItemButton.addEventListener('click', saveItem);
+    }
```

Comment:
- [ ] The requirement is to implement an edit function that can edit the same todo. but in your app when I try to edit any todo, it gets added as a new todo and doesn't update the same todo. kindly consider fixing this bug.
Video Demonstration

https://github.com/SiveMdluli/To_Do_List/assets/96683843/be04f1da-159d-4f58-967a-0aca3680ef4a